presto-native-execution/presto_cpp/main/PrestoTask.cpp

```diff
@@ -583,6 +583,25 @@ protocol::TaskInfo PrestoTask::updateInfoLocked() {
   return str;
 }
 
+std::string PrestoTask::toString() const {
+  // Can this be in a deadlock due to a bug?
+  std::lock_guard<std::mutex> l(mutex);
```

Comment:
- Do we need the lock? 
- If we need the lock, I think we only need for `task->toString()` method. Let's just grab for that and release quickly instead of keeping it for subsequent instructions.

Reply:
Discussed offline, we need the lock because fields in prestotask can change.